### PR TITLE
Feature/ Implement sidebar

### PR DESCRIPTION
**Pull request by JennyDoJD**
- Design
image

- Result (lasted)
image

- Ticket: [Implement Sidebar](https://trello.com/c/G2x16ZA6/43-implement-sidebar)

### CODE REVIEW

**Comment by ghost** [comment]
image
the icon did not align with the center

File: practice/src/javascripts/sidebar.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Identify the links in the sidebar and add/remove the 'active' class when clicked
+let list = document.querySelectorAll('.nav-item');
+
+function activeLink() {
+  list.forEach((item) => item.classList.remove('active'));
+  this.classList.add('active');
+}
+
+list.forEach((item) => item.addEventListener('click', activeLink));
```
Comment (ghost) [inline practice/src/javascripts/sidebar.js]: @JennyDoJD could you apply Event Delegation in this case?

File: practice/src/index.html [hunk 2]
```diff
@@ -81,5 +86,75 @@ <h2 class="secondary-title form-heading-feature">Create new a product</h2>
...
+          <li class="nav-item active">
+            <a href="javascript:void(0)" class="nav-link">
+              <svg width="24" height="24">
+                <use xlink:href="./assets/images/icons/icons.svg#home-icon" />
+              </svg>
+            </a>
+          </li>
+          <li class="nav-item">
+            <a href="javascript:void(0)" class="nav-link">
+              <svg width="24" height="24">
```
Comment (ghost) [inline practice/src/index.html]: I saw the duplicate attributes with, height. Can we improve?

File: practice/src/css/pages/homepage.css [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+.sidebar-header {
+  display: inline-block;
+  padding-top: 24px;
+  border-radius: 12px;
+  padding-left: 10px;
+}
+
+.side-bar-logo {
```
Comment (huyencao-agilityio) [inline practice/src/css/pages/homepage.css]: side-bar class here should be consistent with the other class side bar

File: practice/src/index.html [hunk 4]
```diff
@@ -81,5 +86,75 @@ <h2 class="secondary-title form-heading-feature">Create new a product</h2>
         </div>
       </form>
     </div>
+
+    <!-- Sidebar -->
+    <div class="sidebar">
```
Comment (huyencao-agilityio) [inline practice/src/index.html]: Please douple check HTML5 tag

File: practice/src/javascripts/sidebar.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// Identify the links in the sidebar and add/remove the 'active' class when clicked
+let list = document.querySelectorAll('.nav-item');
```
Comment (huyencao-agilityio) [inline practice/src/javascripts/sidebar.js]: Naming convention?

File: practice/src/javascripts/sidebar.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+// Identify the links in the sidebar and add/remove the 'active' class when clicked
+document.querySelector('.nav-list').addEventListener('click', function (event) {
+  // Check if the clicked element is a nav-link inside a nav-item
+  if (event.target.classList.contains('nav-link')) {
+    // Remove 'active' class from all nav-items
+    document
+      .querySelectorAll('.nav-item')
```
Comment (huyencao-agilityio) [inline practice/src/javascripts/sidebar.js]: Should use querySelectorAll('.nav-item.active') will minimize the number of elements traversed

Comment (JennyDoJD) [reply practice/src/javascripts/sidebar.js]: I have fixed it

File: practice/src/javascripts/sidebar.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+// Identify the links in the sidebar and add/remove the 'active' class when clicked
+document.querySelector('.nav-list').addEventListener('click', function (event) {
+  // Check if the clicked element is a nav-link inside a nav-item
+  if (event.target.classList.contains('nav-link')) {
+    // Remove 'active' class from all nav-items
+    document
```
Comment (ghost) [inline practice/src/javascripts/sidebar.js]: @JennyDoJD I assume that we have only one active item at the same time. Could you consider improving the code?

File: practice/src/javascripts/sidebar.js [hunk 8]
```diff
@@ -2,12 +2,17 @@
...
-      .forEach((item) => item.classList.remove('active'));
+    const clickedNavItem = event.target.closest('.nav-item');
 
-    // Add 'active' class to the parent nav-item of the clicked nav-link
-    event.target.closest('.nav-item').classList.add('active');
+    // Check if the clicked nav-item is already active
+    if (!clickedNavItem.classList.contains('active')) {
+      // Remove 'active' class from all nav-items
+      document
+        .querySelectorAll('.nav-item.active')
```
Comment (ghost) [inline practice/src/javascripts/sidebar.js]: @JennyDoJD why should we need `querySelectorAll` if only one active item at same time?

**Review by canhpham-zb — APPROVED** [review]

**Pull request merged**